portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryMetadataLocalServiceImpl.java

```diff
@@ -63,6 +63,16 @@ public void deleteFileEntryMetadata(long fileEntryId)
 		}
 	}
 
+	@Override
+	public void deleteFileEntryMetadataByExternalReferenceCode(
+			String externalReferenceCode, long companyId)
+		throws PortalException {
+
+		deleteFileEntryMetadata(
+			getDLFileEntryMetadataByExternalReferenceCode(
+				externalReferenceCode, companyId));
```

Comment:
You need to invoke the service methods through the (self) injected `dlFileEntryMetadataLocalService`. This is so that AOP wrappers are invoked.

Reply:
thanks!